src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs

```diff
@@ -418,5 +419,55 @@ private void LoadConfigurations()
                 Directory.CreateDirectory(_destinationCsxPath);
             }
         }       
+
+        private async void CleanupFilesForDeletion()
```

Comment:
async void is not recommended as the best practice for async programming. 
Change it to return Task.

There are lot of articles explaining this. You can also read it here : https://docs.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming